File: backend/api/src/main/java/com/github/ai/leetcodequiz/api/response/GetQuestionnairesResponse.java
package com.github.ai.leetcodequiz.api.response;

import com.github.ai.leetcodequiz.api.QuestionnaireItemDto;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public record GetQuestionnairesResponse(
    @NotNull List<QuestionnaireItemDto> questionnaires
) {
}

File: backend/api/src/main/scala/com/github/ai/leetcodequiz/api/response/GetQuestionnairesResponse.scala
package com.github.ai.leetcodequiz.api.response

import com.github.ai.leetcodequiz.api.QuestionnairesItemDto
import zio.json.{JsonDecoder, JsonEncoder}

case class GetQuestionnairesResponse(
  questionnaires: List[QuestionnairesItemDto]
) derives JsonEncoder,
      JsonDecoder
